src/vs/workbench/contrib/chat/browser/contrib/chatDynamicVariables.ts

```diff
@@ -60,60 +67,120 @@ export class ChatDynamicVariableModel extends Disposable implements IChatWidgetC
 							this.widget.refreshParsedInput();
 						}
 						return null;
-					} else if (Range.compareRangesUsingStarts(ref.range, c.range) > 0) {
+					}
+
+					if (Range.compareRangesUsingStarts(ref.range, c.range) > 0) {
 						const delta = c.text.length - c.rangeLength;
-						return {
-							...ref,
-							range: {
-								startLineNumber: ref.range.startLineNumber,
-								startColumn: ref.range.startColumn + delta,
-								endLineNumber: ref.range.endLineNumber,
-								endColumn: ref.range.endColumn + delta
-							}
+						ref.range = {
+							startLineNumber: ref.range.startLineNumber,
+							startColumn: ref.range.startColumn + delta,
+							endLineNumber: ref.range.endLineNumber,
+							endColumn: ref.range.endColumn + delta,
 						};
+
+						return ref;
 					}
 
 					return ref;
 				}));
 			});
 
-			this.updateDecorations();
+			this.updateVariableDecorations();
 		}));
 	}
 
-	getInputState(): any {
-		return this.variables;
```

Comment:
Why did you delete this? Now navigating input history for dynamic variables doesn't work

Reply:
The interface with optional function tricked me again - TS didn't yell at me when I've changed the name and so I deleted it as unused 😞 Reverted back.

Reply:
It would have been really easy to miss this, so it's another reason not to delete other methods as part of a really large PR. A smaller cleanup PR for things like this would be safer